### ISSUE TITLE
Question of the code

### ISSUE THREAD

**Issue by XuChanghhu** [xujiaommcome/DiCo#4]
Hello,
Its a wonderful work,
Now, we have a question when training this network,
vit_xL = model1(glb_loc_xL, 1)-> Monai UNETR have only one input, whats the meaning of the second input 1?

if the 1 is directly deleted, where the View Recomposition is implemented?
Then, 
_, label_project_xU2 = generate_mip_projectional(xU, vit_xU, 2)  has bugs for differente sizes of Xu and vit_xU.

Hope for your reply.

**Comment by xujiaommcome** [comment]
Hi, thanks for your interest and feedback!

Sorry for the confusion — I directly modified the MONAI UNETR implementation, so the structure of my UNETR might be slightly different from the original MONAI version.
In my code, the forward function is defined as:

def forward(self, x_in, fusion=0):
    x, hidden_states_out = self.vit(x_in)
    if fusion == 1:
        glb_e1, loc_e1 = output.split([1, 4], dim=0)  # [48,48,96]
        last_loc_e1 = self.conv1(blocks2volume(loc_e1))
        last_glb_e1 = F.interpolate(glb_e1, size=[96, 96, 96], mode='trilinear', align_corners=False)
        output = self.conv2(last_loc_e1 + last_glb_e1)
    else:
        output = F.interpolate(output, size=[96, 96, 96], mode='trilinear', align_corners=False)
        output = self.conv3(output)


Here, the second input fusion=1 just controls whether global-local feature recomposition is used inside the modified UNETR.
If you remove this argument, you will lose the view recomposition mechanism.

As for

_, label_project_xU2 = generate_mip_projectional(xU, vit_xU, 2)


you are right — this function assumes that the sizes of xU and vit_xU are the same.
You need to make sure their input sizes are matched before calling generate_mip_projectional.

**Comment by XuChanghhu** [comment]
Thanks for the reply, could u share the modified MONAI UNETR.py?
Cause the definition of self.vit, self.conv1,2,3 is not shown.

**Comment by stap1e** [comment]
You also need give your ViT code for your version, because the latest MONAI can't give a ViT which has keyword argument 'pos_embed'.
 Thanks.
